### ISSUE TITLE
Is there an option to disable data encryption when stored in Redis?

### ISSUE THREAD

**Issue by mingfang** [bungle/lua-resty-session#16]
We are storing session data inside Redis.
Can we disable session data encryption?

**Comment by bungle** [comment]
Not currently. We can add an option for it.

**Comment by bungle** [comment]
I will implement pluggable ciphers soon. Stay tuned.

**Comment by bungle** [comment]
@mingfang,

This is now committed on ciphers branch:
https://github.com/bungle/lua-resty-session/tree/ciphers

This commit contains support for pluggable ciphers:
https://github.com/bungle/lua-resty-session/commit/c89be43d46ce90b74109a5da5b71dd8fe4ce8d32

The code should be backward compatible.

I have now implemented two ciphers:

1. `none` or [resty.session.ciphers.none](https://github.com/bungle/lua-resty-session/blob/ciphers/lib/resty/session/ciphers/none.lua)
2. `aes` or  [resty.session.ciphers.aes](https://github.com/bungle/lua-resty-session/blob/ciphers/lib/resty/session/ciphers/aes.lua)

The second one is the default and it works like before. The `none` cipher is actually what you are looking for. It doesn't encrypt at all.

To use none adapter you need to specify this in Nginx config:

```nginx
set $session_cipher "none";
```

Or alternatively you can pass it to session constructor:

```lua
-- You can use new, open or start as a constructor
local session = require "resty.session".new{ cipher = "none" }
```

Can you please test it? If it works fine, we can release it as 2.1.

Remember that some encoding (base64 or hex) might still happen (cookie adapter encodes the value, but redis doesn't, for example).

Regards
Aapo

**Comment by mingfang** [comment]
I just tested an it works great.
Thank you very much for adding this feature.

**Comment by bungle** [comment]
Okay, thanks! This is now released as a part of Release 2.1.
